src/nanoarrow/common/array.c

```diff
@@ -720,7 +731,40 @@ static int ArrowArrayViewSetArrayInternal(struct ArrowArrayView* array_view,
     }
   }
 
-  // Check the number of buffers
+  if (array_view->storage_type == NANOARROW_TYPE_STRING_VIEW ||
+      array_view->storage_type == NANOARROW_TYPE_BINARY_VIEW) {
+    const int64_t n_buffers = array->n_buffers;
+    const int64_t n_fixed_buffers = 2;
+
+    int64_t n_variadic_buffers = n_buffers - n_fixed_buffers - 1;
+    // Theoretically if the variadic buffers are not used the above subtraction
+    // could yield a negative number
+    n_variadic_buffers = (n_variadic_buffers < 0) ? 0 : n_variadic_buffers;
+
+    array_view->n_variadic_buffers = n_variadic_buffers;
+    // TODO: check malloc failures
+    array_view->variadic_buffer_views =
+        ArrowMalloc(sizeof(struct ArrowBufferView) * n_variadic_buffers);
+
+    for (int64_t i = 0; i < n_variadic_buffers; ++i) {
+      ++buffers_required;
+      array_view->variadic_buffer_views[i].data.data =
+          array->buffers[i + n_fixed_buffers];
+
+      // TODO: why do we need this?
+      // If non-null, set buffer size to unknown.
```

Comment:
This was modeled from the prior loop for fixed buffers, although I don't fully understand why the sizes are set like this